### PR TITLE
Get Tyler ready to register

### PR DESCRIPTION
**Pull request by rafaqz**
This PR includes

- a toml file
- basic doc string on Map and Interpolator
- cleanup and refactor Interpolator code a bit
- `crs` instead of `coordinate_system` and a `GeoInterface.crs` method to match the geospatial ecosystem
- some other minor name changes and formatting of things
- some tests of the Extents and GeoInterface functions

I think we should also split out the code into a few other files, but in the next PR so the move isn't mixed up with the changes.

### CODE REVIEW

**Comment by alex-s-gardner** [comment]
@rafaqz did you want me to look this over?

**Comment by rafaqz** [comment]
Of course ;)

File: src/Tyler.jl [hunk 1]
```diff
@@ -1,29 +1,63 @@
...
+
+-`extent`: the initual extent of the map, as a `GeometryBasics.Rect`
+    or an `Extents.Extent` in the projection of `extent_crs`.
+-`extent_crs`: Any `GeoFormatTypes` compatible crs, the default is wsg84.
+
+# Keywords
+
+-`resolution`: The figure resolution.
+-`figure`: an existing `Makie.Figure` object.
+-`crs`: The providers coordinate reference system.
```
Comment (felixcremer) [inline src/Tyler.jl:47]: What is the relation between the extent_crs and the crs keyword? Is one going to be projected onto the other?

File: src/Tyler.jl [hunk 2]
```diff
@@ -1,29 +1,63 @@
...
-using Tyler.ForInterpolations: tile2positions, Interpolator
 
 const TileImage = Matrix{RGB{N0f8}}
 
+"""
+    Map
+
+    Map(extent, [extent_crs=wgs84]; kw...)
+
+Tylers object, it plots tiles onto a Makie.jl `Axis`,
```
Comment (felixcremer) [inline src/Tyler.jl:34]:
```suggestion
Tylers main object, it plots tiles onto a Makie.jl `Axis`,
```

File: src/Tyler.jl [hunk 3]
```diff
@@ -1,29 +1,63 @@
...
+downloading and plotting more as you zoom and pan.
```
Comment (felixcremer) [inline src/Tyler.jl:35]:
```suggestion
downloading and plotting more tiles as you zoom and pan.
```

File: src/Tyler.jl [hunk 4]
```diff
@@ -1,29 +1,63 @@
...
+
+# Arguments
+
+-`extent`: the initual extent of the map, as a `GeometryBasics.Rect`
```
Comment (felixcremer) [inline src/Tyler.jl:39]:
```suggestion
-`extent`: the initial extent of the map, as a `GeometryBasics.Rect`
```

File: src/Tyler.jl (hunk 1 above)
Comment (rafaqz) [reply src/Tyler.jl:47]: `extent_crs` just lets you enter an extent in a different projection. `crs` is the actual projection of the tiles.

**Review by felixcremer — COMMENTED** [review]
I didn't follow the code changes and mainly proofread the docstrings.
Should we export any of the methods from Tyler or do you expect to always use Tyler.Map?

**Comment by rafaqz** [comment]
Hmm just `Map` is so generic. Leaflet.jl also has `Leaflet.Map`, I like how explicit it is but still concise.

**Comment by rafaqz** [comment]
Ok I'm keen to merge and register this today, any objections?

File: src/Tyler.jl [hunk 5]
```diff
@@ -1,29 +1,63 @@
...
 
-using Makie
-using LinearAlgebra
+import Colors
+import Extents
+import GeoInterface
+import GeometryBasics
+import HTTP
+import ImageMagick
+import LinearAlgebra
```
[the remark this answers is not in the document]
Comment (rafaqz) [reply src/Tyler.jl:9]: LinearAlgebra? I actually don't know

File: src/Tyler.jl [hunk 6]
```diff
@@ -1,29 +1,63 @@
 module Tyler
 
-using Makie
-using LinearAlgebra
+import Colors
```
[the remark this answers is not in the document]
Comment (felixcremer) [reply src/Tyler.jl:3]: But isn't the way to do that nowadays to either explicitely use the functions you need or to explicitely  only get the package into the namespace
```julia
using Colors: Colors
```

[the remark this answers is not in the document]
Comment (rafaqz) [reply src/Tyler.jl:3]: Yeah probably, I can rewrite it like that too

**Comment by SimonDanisch** [comment]
I do actually want to explore moving things to a recipe and figure out a way to do backend specific rendering (e.g. moving things to JS for WGLMakie)... So at least some bigger, breaking changes are lurking at the horizon!

**Comment by rafaqz** [comment]
@SimonDanisch  that sounds pretty good to me

So lets keep it below 1.0 for now... but I think we should get it registered as is, it has 36 stars and a lot of users already.

**Comment by lazarusA** [comment]
mmm.. sure? things usually take more than expected. In the order of months, so being below 1.0 for months is the best option?  I would say just go for it, 1.0 :D

**Pull request merged**